--- decoder_plugins/vorbis/vorbis.c.str	2008-12-26 06:43:55.000000000 +0100
+++ decoder_plugins/vorbis/vorbis.c	2008-12-26 06:44:15.000000000 +0100
@@ -210,7 +210,7 @@
 		char *vorbis_err = vorbis_strerror (res);
 		
 		decoder_error (&data->error, ERROR_FATAL, 0,
-				vorbis_err);
+				"^s", vorbis_err);
 		debug ("ov_open error: %s", vorbis_err);
 		free (vorbis_err);
 
--- interface.c.str	2008-12-26 07:00:50.000000000 +0100
+++ interface.c	2008-12-26 07:01:19.000000000 +0100
@@ -828,7 +828,7 @@
 	
 	send_int_to_srv (CMD_GET_ERROR);
 	err = get_data_str ();
-	error (err);
+	error ("%s",err);
 	free (err);
 }
 
